### PR TITLE
fix(UIShell): add bg color to selected states (NEW)

### LINKED ISSUE #10132: [Bug]: update selected menu item in UI Shell
### Package

carbon-components

### Browser

Chrome

### Package version

current

### Description

Need to update the selected state of in the header menu in the UI Shell. 
From slack comment: https://ibm-studios.slack.com/archives/C046Y0YUD/p1636956609126800

### Screenshots

As-is state:
image

To be state:
image

### Steps to reproduce

- Go to ui shell + header base + navigation 
- Open menu
- Set an item to selected

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by jsehull**
Closes #10132 

Continuation of closed PR (fork error 😬 ): #13385
Please reference this PR for comments and progress

THIS PR finishes CSS details for `HeaderNavigation` and `SubMenu` `selected` states
___

@aagonzales Example of full height border for SideNav in latest fix request 
image

#### Changelog

**New**

- add background colors for both selected states ("selected" and "selected + hover")
- ensured full height pseudo-borders (previously stopped short at border width)

**Changed**

- n/a

**Removed**

- n/a

#### Testing / Reviewing
3. visit Storybook > UI Shell > Header Base w/Navigation to confirm updates for desktop and mobile now display as designed

Please reference the above issues for design details/comments.

___

### CODE REVIEW

**Comment by jsehull** [comment]
recheck

**Comment by tay1orjones** [comment]
recheck

**Review by tw15egan — DISMISSED** [review]
Styles look great, but it seems like it is not changing the parents state if the child is active. I think we can add something like this to `HeaderMenu.js` to handle that

`HeaderMenu.js L205`
```js
    const hasActiveChildren =
      React.Children.map(children, (child) => {
        return child.props.isActive || child.props.isCurrentPage;
      }).length > 0;
```

and then conditionally render the active class on the parent if a) its child has the active state and b) it is not expanded

`HeaderMenus.js L218`
```js
      [`${prefix}--header__menu-item--current`]:
        isActivePage || (hasActiveChildren && !this.state.expanded),
```

With that we get:

**Comment by jsehull** [comment]
Ty for the feedback, will look into this sometime soon

File: packages/react/src/components/UIShell/HeaderMenu.js [hunk 1]
```diff
@@ -193,6 +193,10 @@ class HeaderMenu extends React.Component {
       ...rest
     } = this.props;
 
+    const hasActiveChildren =
+      React.Children.map(children, (child) => {
+        return child.props.isActive || child.props.isCurrentPage;
+      }).length > 0;
```
Comment (francinelucca) [inline packages/react/src/components/UIShell/HeaderMenu.js]: slight performance improvement. you might have to do `.toArray()` before calling `.some`, not sure
```suggestion
    const hasActiveChildren =
      React.Children.some(children, (child) => {
        return child.props.isActive || child.props.isCurrentPage;
      });
```

File: packages/react/src/components/UIShell/HeaderMenu.js [hunk 2]
```diff
@@ -208,6 +212,8 @@ class HeaderMenu extends React.Component {
       // We set the current class only if `isActive` is passed in and we do
       // not have an `aria-current="page"` set for the breadcrumb item
       [`${prefix}--header__menu-item--current`]: isActivePage,
+      [`${prefix}--header__menu-item--current`]:
+        isActivePage || (hasActiveChildren && !this.state.expanded),
```
Comment (francinelucca) [inline packages/react/src/components/UIShell/HeaderMenu.js]: looks like this class is defined twice?

**Review by francinelucca — CHANGES_REQUESTED** [review]
Some refactoring suggestions.

Also looks like Percy is throwing out some changes in the bottom borders of the nav items, wondering if we're cool with those @aagonzales 

https://user-images.githubusercontent.com/40550942/235253112-6cceb73f-76b9-4b8e-84e4-d9d783722eb3.mov

**Comment by aagonzales** [comment]
@francinelucca I think that's fine, looks like the original one was not including the internal border area. It should extend 16px on both sides of the text.

File: packages/react/src/components/UIShell/HeaderMenu.js (hunk 1 above)
Comment (tw15egan) [reply packages/react/src/components/UIShell/HeaderMenu.js]: Updated 👍

**Comment by tw15egan** [comment]
@aagonzales This one should be good to go for the final visual review, check out [this story](https://deploy-preview-13402--carbon-components-react.netlify.app/?path=/story/components-ui-shell--header-base-w-navigation) for testing 👍🏻

**Review by francinelucca — APPROVED** [review]
🚀 LGTM

**Review by aagonzales — APPROVED** [review]
Nice! I think that's all working as intended now 🎉

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**